internal/pkg/build/sources/oci_unpack.go

```diff
@@ -81,7 +81,8 @@ func UnpackRootfs(_ context.Context, srcImage v1.Image, destDir string) (err err
 	}
 
 	// Allow unpacking as non-root
-	if namespaces.IsUnprivileged() {
+	insideUserNS, _ := namespaces.IsInsideUserNamespace(os.Getpid())
+	if namespaces.IsUnprivileged() || insideUserNS {
```

Comment:
This is I think the key change that went into the original PR, but it turns out to be redundant for apptainer because `namespaces.IsUnprivileged()` already covers all the cases that `namespaces.IsInsideUserNamespace(os.Getpid())` covers.  In fact, I don't think apptainer has the problem at all that was being addressed by that PR, that of running under `unshare -r` and `unshare -c`.  They both work for me (the latter on EL9 because it doesn't exist on EL8).